Content.Server/Power/EntitySystems/MachineBatterySystem.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Server.Construction;
+using Content.Server.Power.Components;
+using JetBrains.Annotations;
+
+namespace Content.Server.Power.SMES
+{
+    [UsedImplicitly]
+    public sealed class MachineBatterySystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<MachineBatteryComponent, RefreshPartsEvent>(OnRefreshParts);
+        }
+
+        public void OnRefreshParts(EntityUid uid, MachineBatteryComponent component, RefreshPartsEvent args)
+        {
+            var capacitorRating = args.PartRatings[component.MachinePartPowerCapacity];
+
+            if (TryComp<BatteryComponent>(uid, out var batteryComp))
+            {
+                batteryComp.MaxCharge = MathF.Pow(capacitorRating,component.MachinePartEfficiency) * component.PowerCapacityScaling;
```

Comment:
SetMaxCharge does the same as = though, doesnt it?
`public float MaxCharge { get => _maxCharge; set => SetMaxCharge(value); }`

Reply:
It sets the current charge also so it's capped to max charge. Relevant when downgrading.
https://github.com/space-wizards/space-station-14/blob/2bc332a1b6b8913f55638bad7c59213f21260262/Content.Server/Power/Components/BatteryComponent.cs#L74-L79

Reply:
Isnt that handled by the setter?

https://github.com/space-wizards/space-station-14/blob/2bc332a1b6b8913f55638bad7c59213f21260262/Content.Server/Power/Components/BatteryComponent.cs#L12-L17

Reply:
Yeah you're right, = should work. It will probably be changed to a class method on a system in the future, since components shouldn't hold logic any more.